src/ghga_datasteward_kit/s3_upload/utils.py

```diff
@@ -261,3 +262,13 @@ async def __aexit__(self, exc_t, exc_v, exc_tb):
         # simply reraise unhandled exceptions with unknown upload status
         if exc_v is not None:
             raise exc_v
+
+
+def safe_urljoin(base: str, *paths) -> str:
+    """Join URL parts safely, removing any trailing or leading slashes."""
+    url = base
+    for path in paths:
+        if not url.endswith("/"):
+            url += "/"
+        url = urljoin(url, path.lstrip("/"))
+    return url
```

Comment:
I would rename this function, because the "normal" urljoin is not really "unsafe", and it is not variadic.